Python/sysmodule.c

```diff
@@ -3479,6 +3479,71 @@ make_impl_info(PyObject *version_info)
     return NULL;
 }
 
+static PyObject*
+make_abi_features(void)
+{
+    int res;
+    PyObject *features = PyFrozenSet_New(NULL);
+    if (features == NULL) {
+        goto error;
+    }
+
+#ifdef Py_GIL_DISABLED
+    PyObject *threading_feature = PyUnicode_FromString("free-threading");
+#else
+    PyObject *threading_feature = PyUnicode_FromString("gil-enabled");
+#endif
+    if (threading_feature == NULL) {
+        goto error;
+    }
+    res = PySet_Add(features, threading_feature);
+    Py_DECREF(threading_feature);
+    if (res < 0) {
+        goto error;
+    }
+
+#ifdef Py_DEBUG
+    PyObject *debug = PyUnicode_FromString("debug");
+    if (debug == NULL) {
+        goto error;
+    }
+    res = PySet_Add(features, debug);
+    Py_DECREF(debug);
+    if (res < 0) {
+        goto error;
+    }
+#endif
+
+    PyObject *bitness;
+    switch (PY_SSIZE_T_MAX) {
+    case 0x7FFFFFFFL:
+        bitness = PyUnicode_FromString("32-bit");
+        break;
+    case 0x7FFFFFFFFFFFFFFFL:
+        bitness = PyUnicode_FromString("64-bit");
+        break;
+    default:
+        bitness = Py_NewRef(Py_None);
+        break;
+    }
```

Comment:
Would `0x7FFFFFFFFFFFFFFFL` overflow on some platforms?

```suggestion
    switch (sizeof(Py_ssize_t)) {
        case 4:
            bitness = PyUnicode_FromString("32-bit");
            break;
        case 8:
            bitness = PyUnicode_FromString("64-bit");
            break;
        default:
            bitness = Py_NewRef(Py_None);
            break;
    }
```